analytics_dashboard/core/middleware.py

```diff
@@ -8,10 +8,24 @@
 from django.template.response import TemplateResponse
 
 from core.exceptions import ServiceUnavailableError
+from waffle import flag_is_active
```

Comment:
Since we do want the solution to be upstreamed, I'm going to have to be a bit picky about this change, sorry in advance.

I would prefer if we separate any waffle functionality from here, and move the Waffle Switch/Flag creation to `analytics_dashboard/core/config`. The config can be a directory which contains `waffle.py` or it can be a python file which contains the defined waffle switch/flag. 

And instead of checking `flag_is_active` here, we'd import the Waffle Switch/Flag, and use `SWITCH_VARIABLE.is_enabled()` to check if it is enabled or not.